### PR TITLE
proto/select.proto:init ResponseStream

### PR DESCRIPTION
**Pull request by AndreMouche**
@hicqu @ilovesoup @tiancaiamao @coocood @BusyJay @lamxTyler PTAL

### CODE REVIEW

File: proto/select.proto [hunk 1]
```diff
@@ -130,3 +130,18 @@ message DAGRequest {
...
+        TypeDefault = 0;
+        TypeArrow = 1;
+}
+
+message StreamResponse {
+    optional Error error = 1;
+    optional EncodeType encode_type=2;
+	//Data for all rows
+    optional bytes data = 3 [(gogoproto.customtype) = "github.com/pingcap/tipb/sharedbytes.SharedBytes", (gogoproto.nullable) = false];
+	repeated Error warnings = 4;
```
Comment (coocood) [inline proto/select.proto]: alignment

File: proto/select.proto [hunk 2]
```diff
@@ -130,3 +130,18 @@ message DAGRequest {
...
 }
+
+enum EncodeType {
+        TypeDefault = 0;
+        TypeArrow = 1;
+}
+
+message StreamResponse {
+    optional Error error = 1;
+    optional EncodeType encode_type=2;
```
Comment (coocood) [inline proto/select.proto]: [(gogoproto.nullable) = false];

File: proto/select.proto [hunk 3]
```diff
@@ -130,3 +130,18 @@ message DAGRequest {
 	// It represents which columns we should output.
 	repeated uint32 output_offsets = 5;
 }
+
+enum EncodeType {
+        TypeDefault = 0;
+        TypeArrow = 1;
+}
+
+message StreamResponse {
```
Comment (tiancaiamao) [inline proto/select.proto]: Why we need a `StreamResponse` ?

**Comment by tiancaiamao** [comment]
Any description of this PR? I don't understand why we need introduce a new `StreamResponse` datastruct.

File: proto/select.proto (hunk 3 above)
Comment (AndreMouche) [reply proto/select.proto]: Since some field_type of SelectResposne is redundant in Streaming API while we still need to keep `SelectResponse` in Old API.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**